src/services/data-query.js

```diff
@@ -28,47 +28,63 @@ function QueryService(){
   };
 
   /**
-   *
-   * @param geometry
-   * @param feature_count
-   * @param multilayers
-   * @param condition
-   * @param excludeLayers
+   * @param {{ feature: unknown, feature_count: unknown, filterConfig: unknown, multilayers: boolean, condition: boolean, excludeLayers: unknown[] }}
+   * 
    * @returns {Promise<unknown>}
    */
-  this.polygon = function({feature, feature_count=this.project.getQueryFeatureCount(), filterConfig={}, multilayers=false, condition=this.condition, excludeLayers=[]}={}) {
-    const polygonLayer = excludeLayers[0];
-    const fid = feature.get(G3W_FID);
+  this.polygon = function({
+    feature,
+    feature_count  = this.project.getQueryFeatureCount(),
+    filterConfig   = {},
+    multilayers    = false,
+    condition      = this.condition,
+    excludeLayers  = []
+  } = {}) {
+    const hasExternalLayers = this.getSelectedExternalLayers({ type: "vector" }).length;
```

Comment:
double check, I'm not so sure if this is the correct check 

if we set as in the figure: `hasExternalLayers == false`:

![image](https://user-images.githubusercontent.com/9614886/221601725-5d17a4ce-0568-4c7e-a64c-2c58a4b64cd4.png)

Then I guess this may also be the origin of the fatal exception shown here: https://github.com/g3w-suite/g3w-client/pull/356#issuecomment-1443704286

Reply:
That error https://github.com/g3w-suite/g3w-client/pull/356#issuecomment-1443704286 is triggered at the following line because the `query.layer.getName()` function isn't added to object queries performed on external layers?

https://github.com/g3w-suite/g3w-client/blob/716fdffb9203542e5677e1529872df5ffa4fd8fc/src/components/QueryResults.vue#L239